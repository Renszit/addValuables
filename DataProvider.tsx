import { createContext, useState } from "react";
import { InventoryItem } from "./src/navigation/types";
import MOCK_DATA from "./mockdata.json";

export const DataContext = createContext(null);

export const DataContextProvider = ({ children }) => {
  const [data, setData] = useState<InventoryItem[]>(MOCK_DATA);

  const updateData = (newData: InventoryItem) => {
    setData([newData, ...data]);
  };

  return (
    <DataContext.Provider value={{ data, updateData }}>
      {children}
    </DataContext.Provider>
  );
};
