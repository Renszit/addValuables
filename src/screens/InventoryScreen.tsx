import { useContext, useState } from "react";
import {
  FlatList,
  ScrollView,
  StyleSheet,
  View,
  Dimensions,
} from "react-native";
import { DataContext } from "../../DataProvider";

import InventoryCard from "../components/InventoryCard";
import { Title } from "../components/Title";
import { InventoryItem, RootTabScreenProps } from "../navigation/types";
import { colors } from "../theme/colors";
import { BORDER_RADIUS } from "../theme/constants";

export default function InventoryScreen({
  navigation,
  route,
}: RootTabScreenProps<"Inventory">) {
  const { data } = useContext(DataContext);
  const handleAddButtonPress = () => navigation.navigate("AddItem");

  //TODO: fix type keyExtractor

  return (
    <View style={styles.container}>
      <Title onButtonPress={handleAddButtonPress}>{route.name}</Title>
      <FlatList
        contentContainerStyle={{ paddingBottom: 20 }}
        style={{ height: "100%" }}
        numColumns={2}
        keyExtractor={(item, index) => " key" + index + item.id}
        showsVerticalScrollIndicator={false}
        showsHorizontalScrollIndicator={false}
        data={data}
        renderItem={InventoryCard}
        ListFooterComponent={() => <View style={{ height: 20 }} />}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: "100%",
    paddingHorizontal: 20,
    backgroundColor: colors.background,
    justifyContent: "space-between",
  },
});
