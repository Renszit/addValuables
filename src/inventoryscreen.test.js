import React from "react";
import { render, fireEvent, act } from "@testing-library/react-native";
import { DataContext } from "../DataProvider";
import InventoryScreen from "./screens/InventoryScreen";
import MOCK_DATA from "../mockdata.json";

describe("InventoryScreen", () => {
  const navigation = { navigate: jest.fn() };
  const route = { name: "Inventory" };

  it("renders correctly", () => {
    const { getByText, toJSON } = render(
      <DataContext.Provider value={{ MOCK_DATA }}>
        <InventoryScreen navigation={navigation} route={route} />
      </DataContext.Provider>
    );

    expect(getByText("Inventory")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
