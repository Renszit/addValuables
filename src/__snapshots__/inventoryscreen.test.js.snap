// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F4F3EF",
      "flex": 1,
      "height": "100%",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "height": 42,
        "justifyContent": "space-between",
        "marginTop": 99,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Circular-Bold",
          "fontSize": 34,
          "lineHeight": 42,
        }
      }
    >
      Inventory
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "opacity": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#2D50E6",
            "borderRadius": 14,
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    contentContainerStyle={
      Object {
        "paddingBottom": 20,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "height": 20,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
